### PR TITLE
Add/fivetran amendments

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add handling of a column- and table-blocking scenario to Fivetran interaction. Testing steps in https://github.com/Automattic/newspack-manager-client/pull/16

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by dkoo — CHANGES_REQUESTED** [review]
I'm not sure this is a problem, but when comparing the schema in Fivetran to the data model outlined in the doc, I see some discrepancy.

In Fivetran:

Screen Shot 2022-02-08 at 10 47 54 AM

According to the doc:

Screen Shot 2022-02-08 at 10 48 50 AM

The `member`, `segment`, and `unsubscribe` tables are listed as used in the docs, but are disabled within Fivetran.

File: includes/oauth/class-fivetran-connection.php [hunk 1]
```diff
@@ -96,7 +78,14 @@ public function register_api_endpoints() {
...
-		$connections_stauses = self::process_proxy_response( \wp_safe_remote_get( $url ) );
+		$connections_stauses = self::process_proxy_response(
+			\wp_safe_remote_get(
+				$url,
+				[
+					'timeout' => 30, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout
+				]
+			)
+		);
 		if ( is_wp_error( $connections_stauses ) ) {
```
Comment (miguelpeixe) [inline includes/oauth/class-fivetran-connection.php:89]:
```suggestion
		$connections_statuses = self::process_proxy_response(
			\wp_safe_remote_get(
				$url,
				[
					'timeout' => 30, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout
				]
			)
		);
		if ( is_wp_error( $connections_statuses ) ) {
```

**Review by miguelpeixe — APPROVED** [review]
It's working as described! I was not able to reproduce the issue reported by @dkoo.

Left a couple of requests and should be ok to merge after these small adjustments.

File: includes/oauth/class-fivetran-connection.php [hunk 2]
```diff
@@ -96,7 +78,14 @@ public function register_api_endpoints() {
...
 			return new WP_Error(
```
[the remark this answers is not in the document]
Comment (miguelpeixe) [reply includes/oauth/class-fivetran-connection.php:90]: This is actually not necessary! You skip this one.

**Comment by adekbadek** [comment]
Merged to `release` as a hotfix (https://github.com/Automattic/newspack-plugin/pull/1515) – these changes are now on `master` (https://github.com/Automattic/newspack-plugin/commit/bd76ceab7c9359a47c91a0b25287b41cf0e421b4)

**Pull request closed without merging**